src/PVRIptvData.cpp

```diff
@@ -804,39 +804,74 @@ int PVRIptvData::GetFileContents(std::string& url, std::string &strContent)
+
+  Adapted from https://stackoverflow.com/a/31533119
+  Conversion from UTC date to second, signed 64-bit adjustable epoch version.
+  Written by François Grieu, 2015-07-21; public domain. 
+  */
+
+/*
+  Numbering months from March, all months except the one before that origin repeat
+  with a cycle of 5 months totaling 153 days alternating 31 and 30 days, so that,
+  for any month, and without consideration for leap years, the number of days since
+  the previous February can be computed(within a constant) using addition of an appropriate
+  constant, multiplication by 153 and integer division by 5;
+  the correction in days accounting for the rule for leap year on years multiple - of - 100
+  (which by exception to the multiple - of - 4 rules are non - leap except if multiple of 400)
+  can be computed(within a constant) by addition of an appropriate constant, integer division
+  by 100, multiplication by 3, and integer division by 4;
+  we can compute correction for any epoch using the same formula we use
+  in the main computation, and can do this with a macro so that this correction is computed
+  at compilation time.
+  In this adaption, epoch is fixed to 1970
+  */
+
+
+static long long my_utctime(int year, int mon, int mday, int hour, int min, int sec)
+{
+  int m, y;
+  y = year + 100;
+  m = mon - 1;
+  if (m < 2)
+  {
+    m += 12;
+    --y;
+  }
+  // compute number of days within constant, assuming appropriate origin
+#define MY_MKTIME(Y,M,D) ((long long)Y*365 + Y/4 - Y/100*3/4 + (M+2)*153/5 + D)
+  return (((MY_MKTIME(y, m, mday) - MY_MKTIME((1970 + 99), 12, 1))
+    * 24 + hour) * 60 + min) * 60 + sec;
+#undef MY_MKTIME // this macro is private
+}
+
+
 int PVRIptvData::ParseDateTime(std::string& strDate, bool iDateFormat)
 {
-  struct tm timeinfo;
-  memset(&timeinfo, 0, sizeof(tm));
+  int year = 2000;
+  int mon = 1;
+  int mday = 1;
+  int hour = 0;
+  int min = 0;
+  int sec = 0;
   char sign = '+';
   int hours = 0;
   int minutes = 0;
 
   if (iDateFormat)
-    sscanf(strDate.c_str(), "%04d%02d%02d%02d%02d%02d %c%02d%02d", &timeinfo.tm_year, &timeinfo.tm_mon, &timeinfo.tm_mday, &timeinfo.tm_hour, &timeinfo.tm_min, &timeinfo.tm_sec, &sign, &hours, &minutes);
+    sscanf(strDate.c_str(), "%04d%02d%02d%02d%02d%02d %c%02d%02d",
+      &year, &mon, &mday, &hour, &min, &sec,
+      &sign, &hours, &minutes);
   else
-    sscanf(strDate.c_str(), "%02d.%02d.%04d%02d:%02d:%02d", &timeinfo.tm_mday, &timeinfo.tm_mon, &timeinfo.tm_year, &timeinfo.tm_hour, &timeinfo.tm_min, &timeinfo.tm_sec);
-
-  timeinfo.tm_mon  -= 1;
-  timeinfo.tm_year -= 1900;
-  timeinfo.tm_isdst = -1;
-
-  std::time_t current_time;
-  std::time(&current_time);
-  long offset = 0;
-#ifndef TARGET_WINDOWS
-  offset = -std::localtime(&current_time)->tm_gmtoff;
-#else
-  _get_timezone(&offset);
-#endif // TARGET_WINDOWS
-
-  long offset_of_date = (hours * 60 * 60) + (minutes * 60);
+    sscanf(strDate.c_str(), "%02d.%02d.%04d%02d:%02d:%02d", 
```

Comment:
Not sure this needs to be split over multiple lines, the rest of the code does not do this. But I guess only preference.